File: JS/Tema3_Ejercicio8.js
/*
 * TEMA 3: ESTRUCTURAS CONDICIONALES - EJERCICIO 8
 * ================================================
 * Verificar si un año es bisiesto.
 * 
 * EXPLICACIÓN:
 * - Reglas de año bisiesto:
 *   1. Divisible por 4 y NO por 100 → Bisiesto
 *   2. Divisible por 400 → Bisiesto
 *   3. Otros casos → No bisiesto
 */

function esBisiesto(anio) {
    if (anio % 400 === 0) {
        return true;
    } else if (anio % 100 === 0) {
        return false;
    } else if (anio % 4 === 0) {
        return true;
    } else {
        return false;
    }
}

console.log("=== AÑO BISIESTO ===");
const anos = [2000, 1900, 2004, 2019, 2020, 2100];

anos.forEach(anio => {
    const es = esBisiesto(anio) ? "SÍ" : "NO";
    console.log(`¿${anio} es bisiesto? ${es}`);
});

/*
 * SALIDA ESPERADA:
 * === AÑO BISIESTO ===
 * ¿2000 es bisiesto? SÍ
 * ¿1900 es bisiesto? NO
 * ¿2004 es bisiesto? SÍ
 * ¿2019 es bisiesto? NO
 * ¿2020 es bisiesto? SÍ
 * ¿2100 es bisiesto? NO
 */

File: Java/Tema3_Ejercicio8.java
/*
 * TEMA 3: ESTRUCTURAS CONDICIONALES - EJERCICIO 8
 * =================================================
 * Escriba un programa que determine si un año es bisiesto.
 * Un año es bisiesto si: es divisible por 4 Y (no divisible por 100 OR divisible por 400)
 */

import java.util.Scanner;

public class Tema3_Ejercicio8 {
    public static void main(String[] args) {
        Scanner scanner = new Scanner(System.in);
        
        System.out.print("Digite un año: ");
        int año = scanner.nextInt();
        
        System.out.println("\n=== AÑO BISIESTO ===");
        
        boolean esBisiesto = (año % 4 == 0 && año % 100 != 0) || (año % 400 == 0);
        
        if (esBisiesto) {
            System.out.println("El año " + año + " ES BISIESTO.");
            System.out.println("Febrero tiene 29 días.");
        } else {
            System.out.println("El año " + año + " NO es bisiesto.");
            System.out.println("Febrero tiene 28 días.");
        }
        
        scanner.close();
    }
}

/*
 * SALIDA ESPERADA:
 * Digite un año: 2020
 * 
 * === AÑO BISIESTO ===
 * El año 2020 ES BISIESTO.
 * Febrero tiene 29 días.
 */
